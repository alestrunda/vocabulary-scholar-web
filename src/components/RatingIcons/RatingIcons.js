import React from 'react';
import PropTypes from 'prop-types';
import FloatingActionButton from 'material-ui/FloatingActionButton';
import classNames from 'classnames';
import ThumbDown from 'material-ui/svg-icons/action/thumb-down';
import Thumbup from 'material-ui/svg-icons/action/thumb-up';

const RatingIcons = props => {
  return (
    <div
      className={classNames('rating-icons', props.className)}
      style={props.style}
    >
      <FloatingActionButton
        onClick={props.onWrong}
        backgroundColor="#f00"
        mini={props.mini}
        className="rating-icons__btn rating-icons__btn--wrong"
      >
        <ThumbDown style={{ width: props.iconSize }} />
      </FloatingActionButton>
      <FloatingActionButton
        onClick={props.onCorrect}
        backgroundColor="#0f0"
        mini={props.mini}
        className="rating-icons__btn rating-icons__btn--correct"
      >
        <Thumbup style={{ width: props.iconSize }} />
      </FloatingActionButton>
    </div>
  );
};

RatingIcons.propTypes = {
  className: PropTypes.string,
  style: PropTypes.object,
  onCorrect: PropTypes.func.isRequired,
  onWrong: PropTypes.func.isRequired,
  iconSize: PropTypes.number,
  mini: PropTypes.bool,
};

RatingIcons.defaultProps = {
  iconSize: 20,
  mini: true,
};

export default RatingIcons;
