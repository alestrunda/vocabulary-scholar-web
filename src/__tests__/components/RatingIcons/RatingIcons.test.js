import React from 'react';
import { shallow } from 'enzyme';

import RatingIcons from '../../../components/RatingIcons';

it('renders correctly', () => {
  const onCorrect = jest.fn();
  const onWrong = jest.fn();
  const ratingIcons = shallow(
    <RatingIcons onCorrect={onCorrect} onWrong={onWrong} />
  );
  expect(ratingIcons).toMatchSnapshot();
});

it('handles correct click', () => {
  const onCorrect = jest.fn();
  const onWrong = jest.fn();
  const ratingIcons = shallow(
    <RatingIcons onCorrect={onCorrect} onWrong={onWrong} />
  );
  ratingIcons.find('.rating-icons__btn--correct').simulate('click');
  expect(onCorrect).toHaveBeenCalled();
});

it('handles wrong click', () => {
  const onCorrect = jest.fn();
  const onWrong = jest.fn();
  const ratingIcons = shallow(
    <RatingIcons onCorrect={onCorrect} onWrong={onWrong} />
  );
  ratingIcons.find('.rating-icons__btn--wrong').simulate('click');
  expect(onWrong).toHaveBeenCalled();
});
