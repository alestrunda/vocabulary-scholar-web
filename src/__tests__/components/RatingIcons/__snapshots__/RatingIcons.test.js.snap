// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <RatingIcons
    iconSize={20}
    mini={true}
    onCorrect={[MockFunction]}
    onWrong={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <FloatingActionButton
          backgroundColor="#f00"
          className="rating-icons__btn rating-icons__btn--wrong"
          disabled={false}
          mini={true}
          onClick={[MockFunction]}
          secondary={false}
          zDepth={2}
        >
          <ActionThumbDown
            style={
              Object {
                "width": 20,
              }
            }
          />
        </FloatingActionButton>,
        <FloatingActionButton
          backgroundColor="#0f0"
          className="rating-icons__btn rating-icons__btn--correct"
          disabled={false}
          mini={true}
          onClick={[MockFunction]}
          secondary={false}
          zDepth={2}
        >
          <ActionThumbUp
            style={
              Object {
                "width": 20,
              }
            }
          />
        </FloatingActionButton>,
      ],
      "className": "rating-icons",
      "style": undefined,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "backgroundColor": "#f00",
          "children": <ActionThumbDown
            style={
              Object {
                "width": 20,
              }
            }
          />,
          "className": "rating-icons__btn rating-icons__btn--wrong",
          "disabled": false,
          "mini": true,
          "onClick": [MockFunction],
          "secondary": false,
          "zDepth": 2,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "style": Object {
              "width": 20,
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "backgroundColor": "#0f0",
          "children": <ActionThumbUp
            style={
              Object {
                "width": 20,
              }
            }
          />,
          "className": "rating-icons__btn rating-icons__btn--correct",
          "disabled": false,
          "mini": true,
          "onClick": [MockFunction],
          "secondary": false,
          "zDepth": 2,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "style": Object {
              "width": 20,
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <FloatingActionButton
            backgroundColor="#f00"
            className="rating-icons__btn rating-icons__btn--wrong"
            disabled={false}
            mini={true}
            onClick={[MockFunction]}
            secondary={false}
            zDepth={2}
          >
            <ActionThumbDown
              style={
                Object {
                  "width": 20,
                }
              }
            />
          </FloatingActionButton>,
          <FloatingActionButton
            backgroundColor="#0f0"
            className="rating-icons__btn rating-icons__btn--correct"
            disabled={false}
            mini={true}
            onClick={[MockFunction]}
            secondary={false}
            zDepth={2}
          >
            <ActionThumbUp
              style={
                Object {
                  "width": 20,
                }
              }
            />
          </FloatingActionButton>,
        ],
        "className": "rating-icons",
        "style": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "backgroundColor": "#f00",
            "children": <ActionThumbDown
              style={
                Object {
                  "width": 20,
                }
              }
            />,
            "className": "rating-icons__btn rating-icons__btn--wrong",
            "disabled": false,
            "mini": true,
            "onClick": [MockFunction],
            "secondary": false,
            "zDepth": 2,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "style": Object {
                "width": 20,
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "backgroundColor": "#0f0",
            "children": <ActionThumbUp
              style={
                Object {
                  "width": 20,
                }
              }
            />,
            "className": "rating-icons__btn rating-icons__btn--correct",
            "disabled": false,
            "mini": true,
            "onClick": [MockFunction],
            "secondary": false,
            "zDepth": 2,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "style": Object {
                "width": 20,
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;
